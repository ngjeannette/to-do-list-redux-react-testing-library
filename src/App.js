import './App.css';
import React from 'react';
import { ToDo } from './features/Todo/Todo';

function App() {

  return (
    <div className="App">
      <div className="App-header">
        <ToDo />
      </div>
    </div>
  );
}

export default App;