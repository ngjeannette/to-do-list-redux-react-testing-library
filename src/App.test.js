import React from 'react';
import { render as rtlRender, screen } from '@testing-library/react'
import { Provider } from 'react-redux';
import appStore from './app/store';
import App from './App';
import userEvent from '@testing-library/user-event';

const initialState = {
  todo: {
    value: [],
  }
};
function render(
  ui,
  {
    initialState,
    store = appStore,
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
};

describe('To Do Testing', () => {
  beforeEach(() => {
    render(<App />);
  })

  // 1) test to render h1 tags UI
  test('renders h1 tag "To Do Test" ', () => {
    expect(screen.getByRole('heading', { level: 1 }).textContent).toBe('To Do Test');
  });

  // 2) test to render submit will add onto list
  test('type word "React" then submit will add onto list', () => {
    const input = screen.getByPlaceholderText('Feed the 🐱');
    const submit = screen.getByText(/submit/i)
    userEvent.type(input, 'React');
    userEvent.click(submit);
    expect(screen.getByText(/React/i).textContent).toBe('React');
  });

  // 3) test click of check will check checkbox, and click again will remove checkbox... checking checkbox checks checkbox :D
  test('click of check will toggle checkbox true', () => {
    const checkbox = screen.getByRole('checkbox', { name: /checkbox/i });
    userEvent.click(checkbox);
    expect(checkbox.checked).toBe(true);
  });

  // 4) test click of check update complete total + 1
  test('check of checkbox updates complete', () => {
    const checkbox = screen.getByRole('checkbox', { name: /checkbox/i });
    userEvent.click(checkbox);
    const total = screen.getByTestId('total');
    expect(total.textContent).toBe("1");
  });
  // 5) test click of check update complete total - 1
  test('uncheck of checkbox updates complete', () => {
    const checkbox = screen.getByRole('checkbox', { name: /checkbox/i });
    userEvent.click(checkbox);
    userEvent.click(checkbox);
    const total = screen.getByTestId('total');
    expect(total.textContent).toBe("0");
  });

  // 6) snapshot
  test('snapshot', () => {
    const { asFragment } = render(<Provider store={appStore}><App /></Provider>);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
})