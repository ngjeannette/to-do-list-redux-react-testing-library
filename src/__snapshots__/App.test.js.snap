// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To Do Testing snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="App-header"
    >
      <div
        class="App"
      >
        <div
          class="App-header"
        >
          <div>
            <h1
              class="has-text-black is-size-1"
            >
              To Do Test
            </h1>
            <div
              class="input-container"
            >
              <input
                aria-label="textbox"
                class="input"
                maxlength="20"
                placeholder="Feed the 🐱"
                type="text"
                value=""
              />
              <button
                aria-label="submit"
                class="button is-link"
                id="submit"
              >
                 Submit
              </button>
            </div>
            <nav
              class="panel"
            >
              <p
                class="panel-heading"
              >
                Keeping in Track
              </p>
              <div
                class="checklist"
              >
                <div
                  class="panel-block"
                >
                  <label
                    class="checkbox"
                    for="item-0"
                  >
                    <input
                      aria-label="checkbox"
                      id="item-0"
                      type="checkbox"
                    />
                    React
                  </label>
                </div>
              </div>
              <div
                class="panel-block total"
              >
                <span
                  class="is-outlined is-fullwidth"
                >
                   Completed: 
                </span>
                <span
                  data-testid="total"
                  id="total"
                >
                  0
                </span>
                <span>
                  /1
                </span>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
